content/en/docs/generating-objects/instantiate-methods.md

```diff
@@ -0,0 +1,644 @@
+            constructor() {
+                parameter<String>()
+            }
+        }
+        .sample()
+
+    then(genericObject).isNotNull()
+    then(genericObject.value).isNotNull()
+}
+
+{{< /tab >}}
+{{< /tabpane>}}
+
+You can specify to use the constructor with the actual type when working with generic objects.
+
+### Using Constructors with Nested Objects
+In scenarios involving nested objects, where you wish to specify the creation of both objects using their constructors, you can designate each type and specify the constructor to be used.
+
+For example, consider the `ProductList` class that uses the `Product` class:
+
+{{< tabpane persist=false >}}
+{{< tab header="Java" lang="java">}}
+
+@Value
+public class ProductList {
+    String listName;
+    List<Product> list;
+
+    public ProductList(List<Product> list) {
+        this.listName = "defaultProductListName";
+        this.list = list;
+    }
+}
+
+{{< /tab >}}
+{{< tab header="Kotlin" lang="kotlin">}}
+
+class ProductList(val listName: String, val list: List<Product>) {
+    constructor(list: List<Product>) : this("defaultProductListName", list)
+}
+
+{{< /tab >}}
+{{< /tabpane>}}
+
+You can specify to use a certain constructor for both `ProductList` and `Product` with their constructors like the following:
+
+{{< tabpane persist=false >}}
+{{< tab header="Java" lang="java">}}
+
+@Test
+void test() {
+    ProductList productList = fixtureMonkey.giveMeBuilder(ProductList.class)
+        .instantiate(
+            ProductList.class,
+            constructor()
+                .parameter(new TypeReference<List<Product>>() {}, "list")
+        )
+        .instantiate(
+            Product.class,
+            constructor()
+                .parameter(long.class)
+                .parameter(long.class)
+                .parameter(new TypeReference<List<String>>(){})
+        )
+        .size("list", 1)
+        .sample();
+
+    then(productList.getListName()).isEqualTo("defaultProductListName");
+    then(productList.getList()).hasSize(1);
+    then(productList.getList().get(0).getProductName()).isEqualTo("defaultProductName");
+}
+
+{{< /tab >}}
+{{< tab header="Kotlin" lang="kotlin">}}
+
+@Test
+fun test() {
+    val productList = fixtureMonkey.giveMeBuilder<ProductList>()
+        .instantiateBy {
+            constructor<ProductList> {
+                parameter<List<Product>>("list")
+            }
+            constructor<Product> {
+                parameter<Long>()
+                parameter<Long>()
+                parameter<List<String>>()
+            }
+        }
+        .size("list", 1)
+        .sample()
+
+    then(productList.listName).isEqualTo("defaultProductListName")
+    then(productList.list).hasSize(1)
+    then(productList.list[0].productName).isEqualTo("defaultProductName")
+}
+
+{{< /tab >}}
+{{< /tabpane>}}
+
+{{< alert icon="💡" text="It is also possible to mix both constructor and factory method approaches within the instantiate method." />}}
```

Comment:
It can lead to confusion. It is better to clarify that they can be mixed but not in the same property.